src/fabric.c

```diff
@@ -610,29 +610,22 @@ int DEFAULT_SYMVER_PRE(fi_getinfo)(uint32_t version, const char *node,
 			continue;
 		}
 
-		if (ofi_is_util_prov(prov->provider)) {
-			if (flags & OFI_CORE_PROV_ONLY) {
-				FI_INFO(&core_prov, FI_LOG_CORE,
-					"Need core provider, skipping util %s\n",
-					prov->provider->name);
-				continue;
-			}
+		if (ofi_is_util_prov(prov->provider) &&
+		    (flags & OFI_CORE_PROV_ONLY)) {
+			FI_INFO(&core_prov, FI_LOG_CORE,
+			       "Need core provider, skipping util %s\n",
+			       prov->provider->name);
+			continue;
+		}
 
-			if (util_len && util_name &&
-			    strncasecmp(util_name, prov->provider->name, util_len)) {
-				FI_INFO(&core_prov, FI_LOG_CORE,
-					"Want util %s, skipping %s\n",
-					util_name, prov->provider->name);
+		if (util_len && util_name) {
+			assert(!(flags & OFI_CORE_PROV_ONLY));
+			if (strncasecmp(util_name, prov->provider->name, util_len))
```

Comment:
I guess that combined check can help us to reach comparing of `core_name` with `provider->name` that is below in `else-if` statement 
I propose the following:
`if (util_len && util_name &&
     strncasecmp(util_name, prov->provider->name, util_len)) {`